encoding/dablock.go

```diff
@@ -0,0 +1,47 @@
+package encoding
+
+import (
+	"encoding/binary"
+	"errors"
+	"math/big"
+)
+
+// DABlock represents a Data Availability Block.
+type DABlock struct {
+	BlockNumber     uint64
+	Timestamp       uint64
+	BaseFee         *big.Int
+	GasLimit        uint64
+	NumTransactions uint16
+	NumL1Messages   uint16
+}
```

Comment:
all fields of any type that we're expecting to handle solely via interface should be private and exposed via interface, otherwise they're not accessible without casting into the respective type

Reply:
fixed in [1594e0f](https://github.com/scroll-tech/da-codec/pull/25/commits/1594e0fc92e7b345363ec6a7d677a50887c9d002).